sphinx/writers/html.py

```diff
@@ -450,7 +450,7 @@ def visit_literal_block(self, node: Element) -> None:
 
         highlighted = self.highlighter.highlight_block(
             node.rawsource, lang, opts=opts, linenos=linenos,
-            location=(self.builder.current_docname, node.line), **highlight_args
+            location=(node.source, node.line), **highlight_args
```

Comment:
As I commented at https://github.com/sphinx-doc/sphinx/issues/8093#issuecomment-672088541, I think using `location=node` is better. It tries to get a source and line in the best effort (ex. it tries to trace parent node if `node.source` and `node.line` is not set). No reason to use `node.source` and `node.line` directly.

Reply:
I'll give a try